datastore/api/snippets.py

```diff
@@ -0,0 +1,820 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+from collections import defaultdict
+import datetime
+from pprint import pprint
+import time
+
+import gcloud
+from gcloud import datastore
+
+
+def incomplete_key(client):
+    # [START incomplete_key]
+    key = client.key('Task')
+    # [END incomplete_key]
+
+    return key
+
+
+def named_key(client):
+    # [START named_key]
+    key = client.key('Task', 'sample_task')
+    # [END named_key]
+
+    return key
+
+
+def key_with_parent(client):
+    # [START key_with_parent]
+    key = client.key('TaskList', 'default', 'Task', 'sample_task')
+    # [END key_with_parent]
+
+    return key
+
+
+def key_with_multilevel_parent(client):
+    # [START key_with_multilevel_parent]
+    key = client.key(
+        'User', 'alice',
+        'TaskList', 'default',
+        'Task', 'sample_task')
+    # [END key_with_multilevel_parent]
+
+    return key
+
+
+def basic_entity(client):
+    # [START basic_entity]
+    task = datastore.Entity()
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+    # [END basic_entity]
+
+    return task
+
+
+def entity_with_parent(client):
+    # [START entity_with_parent]
+    key_with_parent = client.key(
+        'TaskList', 'default', 'Task', 'sample_task')
+
+    task = datastore.Entity(key=key_with_parent)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+    # [END entity_with_parent]
+
+    return task
+
+
+def properties(client):
+    # [START properties]
+    task = datastore.Entity()
+    task.update({
+        # String
+        'type': 'Personal',
+        'description': 'Learn Cloud Datastore',
+        # Date
+        'created': datetime.datetime.utcnow(),
+        # Boolean
+        'done': False,
+        # Integer
+        'priority': 4,
+        # Float
+        'percent_complete': 10.5,
+    })
+    # [END properties]
+
+    return task
+
+
+def array_value(client):
+    # [START array_value]
+    task = datastore.Entity()
+    task.update({
+        'tags': [
+            'fun',
+            'programming'
+        ],
+        'collaborators': [
+            'alice',
+            'bob'
+        ]
+    })
+    # [END array_value]
+
+    return task
+
+
+def upsert(client):
+    # [START upsert]
+    complete_key = client.key('Task', 'sample_task')
+
+    task = datastore.Entity(key=complete_key)
+
+    task.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    client.put(task)
+    # [END upsert]
+
+    return task
+
+
+def insert(client):
+    # [START upsert]
```

Comment:
This **also** says `START upsert`.

FWIW we don't even implement `insert` ([yet][1]).

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/issues/599